datahub-web-react/src/app/entity/dataset/DatasetEntity.tsx

```diff
@@ -253,6 +255,11 @@ export class DatasetEntity implements Entity<Dataset> {
     renderSearch = (result: SearchResult) => {
         const data = result.entity as Dataset;
         const genericProperties = this.getGenericEntityProperties(data);
+        const snippet = result.matchedFields[0].value.includes('urn:li:tag') ? (
+            <TagSummary urn={result.matchedFields[0].value} />
+        ) : (
+            <TermSummary urn={result.matchedFields[0].value} />
+        );
```

Comment:
I don't think it's guaranteed that if this is not a tag then it is a term (in fact I don't think we surface terms here at all?)

I would say if it includes this `urn:li:tag` then use the tags summary, then do another check if it contains `urn:li:glossaryTerm` and use the terms summary, then if neither of those match, use `<b>{result.matchedFields[0].value}</b>` as it was before

Reply:
Fixed and committed!!